### PR TITLE
[FIX]: Fix PR expand Feature 

### PR DESCRIPTION
**Pull request by VipinDevelops**
## Issue(s)
- #136 
- #135 

## Acceptance Criteria Fulfillment

- [x] Fix the issue of loading multiple buttons after obtaining the pull link by using IPreMessageSentExtend.
- [x] Add a proper check for the number of PRs in a message and render a different button to handle each PR, with a maximum of 3 links.
- [x] Add the "Approve" button in the PR details modal to quickly approve changes.

## Proposed Changes (including videos or screenshots)
[Screencast from 2024-03-06 22-54-35.webm](https://github.com/RocketChat/Apps.Github22/assets/99081689/6dae8829-d1dd-423c-9bc8-2501b55de078)

### CODE REVIEW

File: github/handlers/GithubPRlinkHandler.ts [hunk 1]
```diff
@@ -1,62 +1,45 @@
...
+    extend: IMessageExtender
+) {
+    const githubPRLinkRegex = /https?:\/\/github\.com\/(\S+)\/(\S+)\/pull\/(\d+)/g;
+    const text = message.text!;
+    let prLinkMatches: RegExpExecArray | null;
+    const matches: RegExpExecArray[] = [];
+
+    while ((prLinkMatches = githubPRLinkRegex.exec(text)) !== null) {
+        matches.push(prLinkMatches);
+    }
```
Comment (samad-yar-khan) [inline github/handlers/GithubPRlinkHandler.ts:21]: This while loop is not making sense

**Review by samad-yar-khan — CHANGES_REQUESTED** [review]

**Comment by samad-yar-khan** [comment]
@VipinDevelops Great improvement over the previous code !

File: github/handlers/GithubPRlinkHandler.ts (hunk 1 above)
Comment (VipinDevelops) [reply github/handlers/GithubPRlinkHandler.ts:21]: We are using this loop to get all the possible PR links from the message.

This while loop iterates through the matches of the regular expression in the message text. It uses `exec()` to find the next match in the text. As long as there is a match `(!== null)`, it adds the match to the matches array.

**Comment by VipinDevelops** [comment]
> @VipinDevelops Great improvement over the previous implementation, user experience feels cleaner !

I am glad you like this approach it took me some time to figure out the best alternative for this.

File: github/GithubApp.ts [hunk 2]
```diff
@@ -95,7 +83,10 @@ export class GithubApp extends App implements IPreMessageSentExtend,IPostMessage
         if (await hasGitHubCodeSegmentLink(message)) {
             await handleGitHubCodeSegmentLink(message, read, http, message.sender, message.room, extend);
         }
-        
+        if (await hasGithubPRLink(message)) {
+            await handleGithubPRLinks(message, read, http, message.sender, message.room, extend);
+        }
```
Comment (samad-yar-khan) [inline github/GithubApp.ts:88]: @VipinDevelops  do you think adding more of these checks in checkPreMessageSentExtend will hinder performance ?

**Review by samad-yar-khan — APPROVED** [review]
LGTM, Thanks @VipinDevelops

**Pull request merged**